microsoft-edge/devtools-guide-chromium/evaluate-performance/selector-stats.md

```diff
@@ -0,0 +1,53 @@
+
+The **Performance** tool highlights long tasks with red triangles to indicate work on the main thread that takes too long and has slow performance.  
+
+![Long Task indicators for events on the main thread with slow performance.](../media/evaluate-performance-performance-animation-frame-fired.msft.png)
+
+In your performance recordings, some of these long-running tasks may be **Recalculate Style** events.  These events track the time it takes for the browser to iterate through DOM elements on a page, find all of the CSS style rules that match a given element, and then compute the element's actual style based on these rules.  These styles need to be recalculated whenever the applicabiliy of CSS rules may have changed, like when:
+
+- Elements are added or removed from the DOM
+- An element's attributes, like `class` or `id`, were changed
+- User input, like mouse move or focus change, which can affect `:hover` rules
+
+When you have long running **Recalculate Style** events, you can use the **Selector Stats** feature to understand which of your CSS selectors are taking up the most time and resulting in the slow performance.
+
+## How to use Selector Stats
+
+To see **Selector Stats** during **Recalculate Style** events, complete the following steps.
+
+1. In the **Performance** tool, click **Capture settings** (![Capture settings.](../media/capture-settings-icon.msft.png)).
+
+1. Click **Enable advanced rendering instrumentation (slow)** to turn the setting on. Note that this setting adds more overhead for performance recordings so you should only leave it turned on if you need to record **Selector Stats** and other rendering information.
+ 
+1. Click **Record** and run the scenario you want to improve for your website or app.
+
+1. Click **Stop**.
+
+1. Now find a **Recalculate Style** event. In the bottom section of the **Performance** tool, select **Selector Stats**.
+
+   ![Selector Stats in the Performance tool.](images/performance-tool-selector-stats.png)
+ 
+The **Selector Stats** section of the **Performance** tool shows you a table of selectors, the amount of time the browser spent calculating the style rules, and the number of elements the selector matched.
+
+## Analyze Selector Stats
+
+You can sort the data in the **Selector Stats** table in ascending or descending order for any of the columns by clicking the column header.  For example, sort the **Elapsed (µs)** column in descending order to see which selectors take up the most time in the **Recalculate Style** event you have selected.
+
+As a general rule, if you find a selector that took the browser a long time to calculate and was matched many times, that would be a good candidate to try and improve.  Try simplifying the selector or making it more specific so that fewer elements match to it.  As a result of making these improvements, repeat the steps from the previous section, and you should observe the time the browser spends in the **Recalculate Style** event decrease.
```

Comment:
```suggestion
As a general rule, if you find a selector that took the browser a long time to calculate and was matched many times, that would be a good candidate to try and improve.  Try simplifying the selector or making it more specific so that fewer elements match to it.  As a result of making these improvements, repeat the steps from the previous section, and you should observe that the time the browser spends in the **Recalculate Style** event decrease.
```
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/t/that-vs-which - 
"Include _that_ even if the sentence is clear without it. It helps to clarify the sentence structure."

Reply:
I'm unclear whether "making it more specific" is the right thing to say here. People might interpret it as: "Oh I need to add more parents to my selector for it to be more specific", which might also cause the browser to do more work since it has to go up the chain of ancestors and check more things that it should.

Saying how to improve selectors is dangerous business, and I think our docs should stay away from this as much as we can. We provide a tool, we tell people to measure before and after, and that's great. I don't think we should necessarily tell them how to fix their selectors because that will highly depend on their particular case.

```suggestion
As a general rule, if you find a CSS selector that took the browser a long time to calculate and was matched many times, that's a good candidate to try and improve.  Try to change your selector so it requires less time to calculate and matches fewer elements on the page.  How to improve your CSS selectors depend on your particular use case. Repeat the steps from the previous section to confirm that your changes helped decrease the **Recalculate Style** event duration.
```